lopezMidterm/src/components/Checkout.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from '../store';
+import { removeFromCart, clearCart } from '../actions/cartActions';
+import axios from 'axios';
+
+interface Product {
+  id: number;
+  title: string;
+  price: number;
+}
+
+interface CartItem extends Product {
+  quantity: number;
+}
+
+const Checkout: React.FC = () => {
+  const cartItems = useSelector((state: RootState) => state.cart.items);
+  const dispatch = useDispatch();
+  const [totalPrice, setTotalPrice] = useState(0);
+
+  useEffect(() => {
+    const fetchPrices = async () => {
+      try {
+        const response = await axios.get('/products');
```

Comment:
Should be using a custom hook here instead of axios.get directly in the component.